### PR TITLE
[Do not merge] Keynote demo tweaks

### PR DESCRIPTION
**Pull request by detiber**
Tracking PR for changes made to support or improve deployment for the keynote demo.

Changes so far:
- hack workaround for disabling openshfit_no_proxy_internal_hostnames 
- Addition of openshift_projects role, allowing the ability to set default node selectors on a project level
- Remove injecting of credentials file for router
- Tweaks to router deployment
- Move registry config into openshift_hosted role, remove injecting of credentials file
- Introduction of openshift_hosted_infra_selector to set the selector for openshift_hosted_*_selector with one variable

### CODE REVIEW

**Comment by detiber** [comment]
@abutcher could use an eye on these changes as you get a chance, I'm sure I'm leaving plenty of rough edges along the way.

File: playbooks/common/openshift-cluster/evaluate_groups.yml [hunk 1]
```diff
@@ -36,6 +36,7 @@
       ansible_ssh_user: "{{ g_ssh_user | default(omit) }}"
       ansible_become: "{{ g_sudo | default(omit) }}"
     with_items: g_all_hosts | default([])
+    changed_when: false
```
Comment (abutcher) [inline playbooks/common/openshift-cluster/evaluate_groups.yml]: Nice!

File: roles/openshift_hosted/tasks/router.yml [hunk 2]
```diff
@@ -25,41 +25,41 @@
...
   register: openshift_hosted_router_nodes_json
   changed_when: false
-  when: openshift.hosted.router.replicas | default(None) == None
+  when: openshift.hosted.router.replicas | default(none) is none
 
-- name: Collect nodes matching router selector
-  set_fact:
-    openshift_hosted_router_nodes: >
-      {{ (openshift_hosted_router_nodes_json.stdout|from_json)['items']
-         | oo_oc_nodes_matching_selector(openshift.hosted.router.selector) }}
```
Comment (abutcher) [inline roles/openshift_hosted/tasks/router.yml]: Nice! I think we can delete this filter?

Comment (detiber) [inline roles/openshift_hosted/tasks/router.yml]: possibly, I didn't test any backwards compat.

**Comment by detiber** [comment]
@abutcher rebased and tried to do some targeted squashing to clean things up a bit.

**Comment by detiber** [comment]
aos-ci-test

**Comment by detiber** [comment]
aos-ci-test

**Comment by tbielawa** [comment]
@detiber please close if you don't need this PR any more

**Pull request closed without merging**